### PR TITLE
add graphql codegen

### PR DESCRIPTION
**Pull request by narthur**

### CODE REVIEW

**Comment by narthur** [comment]
@mattleff FYI this PR is now ready for review.

File: getter-plugin.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module.exports = {
+	plugin: (schema, documents, config, info) => {
+		const result = documents
+			.map((doc) => {
+				return doc.document.definitions
+					.filter((def) => {
+						return (
+							def.kind !== 'FragmentDefinition' && def?.operation !== 'mutation'
```
Comment (mattleff) [inline getter-plugin.js:8]: We don't need `def?.operation` since `def` should always be defined.

File: src/jest.spec.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { when } from 'jest-when';
```
Comment (mattleff) [inline src/jest.spec.ts:1]: Do we need this file?

**Review by mattleff — APPROVED** [review]
Great work! It's really nice to have those types generated :shipit: 🍹

File: src/jest.spec.ts (hunk 2 above)
Comment (narthur) [reply src/jest.spec.ts:1]: Yes, I'd like to keep this around. If an API is confusing enough that I feel the need to write a test to figure it out, I like to keep the test for documentation purposes.

**Pull request merged**